sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -876,6 +878,49 @@ class DataTypeSuite extends SparkFunSuite {
       }
   }
 
+  test("string field with invalid collation name") {
+    val collationProviders = Seq("spark", "icu")
+    collationProviders.foreach { provider =>
+      val json =
+        s"""
+           |{
+           |  "type": "struct",
+           |  "fields": [
+           |    {
+           |      "name": "c1",
+           |      "type": "string",
+           |      "nullable": false,
+           |      "metadata": {
+           |        "${DataType.COLLATIONS_METADATA_KEY}": {
+           |          "c1": "$provider.INVALID"
```

Comment:
I think it would also make sense for this conf to bypass the case when the provider is uknown  as well